tools/sigh.ts

```diff
@@ -430,6 +431,23 @@ function webpack(): boolean {
   return result.success;
 }
 
+function pipeshellWebDist(): boolean {
+  const result = saneSpawnWithOutput('cd shells/pipes-shell/web/deploy && ./deploy.sh', []);
```

Comment:
Relies on bash?  Probably breaks on windows?  Maybe just replace the entire shell script with node operations..



```
mkdir dist    # fs.mkdir(..._)

# sources
cp -fR source/index.html dist/   # install ncp module  https://www.npmjs.com/package/ncp

# worker build
cp -fR ../../../lib/build/worker.js dist/   # ncp again
# particles
mkdir dist/particles  # fs.mkdir()
#
mkdir dist/particles/Common  # fs.mkdir()
cp -fR ../../../../particles/Common/Description.schema dist/particles/Common  # ncp again
#
mkdir dist/particles/PipeApps  # fs.mkdir()
cp -fR ../../../../particles/PipeApps/* dist/particles/PipeApps  # ncp
# collate sources
echo packing...
npx webpack   # we should use webpack api, https://webpack.js.org/api/node/ 
echo done.   # console.log
```

Reply:
Or add a Windows OS switch and cmd/bat/powershell (lol) script.

We have been using cross-tool in the npm bits to help with some of this but it'd be nice to tie ourselves less to node rather than more. Even cross-tool hasn't been magic happiness.

We could file an issue for the Windows version and just test for Linux and skip with a warning for now.

Reply:
Fwiw, I use Git Bash when I'm on Windows (hence the bash-style scripts).